scripts/harbor-init/harbor-init.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+const (
+	// Harbor API version
+	HARBOR_API_ENDPOINT = "/api/v2.0"
+)
+
+type Login struct {
+	Url      string
+	Username string
+	Password string
+}
+
+type Projects []struct {
+	Name string `json:"name"`
+}
+
+type ProjectRobots []struct {
+	Name        string `json:"name"`
+	Id          int    `json:"id"`
+	Permissions []struct {
+		Namespace string `json:"namespace"`
+	} `json:"permissions"`
+}
+
+type ResponseRobot struct {
+	Id int `json:"id"`
+}
+
+func request(method string, login Login, endpoint string, body []byte) (*http.Response, []byte, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest(method, login.Url+endpoint, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, nil, err
+	}
+	req.SetBasicAuth(login.Username, login.Password)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil, err
+	}
+	return resp, respBody, nil
```

Comment:
suggestion(non-blocking): If the only reason you are doing this is to deserialize the json from the body... then there is an easier way:

```go
var deserializeIntoHere SomeType
err = json.NewDecoder(res.Body).Decode(&deserializeIntoHere)
```

So this method can just return the response and you can deserialize outside of this function if you want. But this is also fine - just fyi. If you want ignore this.

Reply:
not done, I am not sure how to achieve what I need (i.e. deserialize into different types)

Reply:
ok np.